nomad/bin/deploy

```diff
@@ -1,30 +1,87 @@
-then
-  echo "Unable to connect to nomad-host-prod1. Ensure you're on VPN."
+# --- Pre-flight Checks ---
+
+# 1. Validate Arguments
+if [[ -z "${APP}" ]]; then
+  echo "Error: Missing Application argument." >&2 # Send errors to stderr
+  echo "Usage: [BRANCH=<branch>] [REPO=<repo>] $0 <app> <env>" >&2
+  exit 1 # Exit with non-zero status on error
+fi
+
+if [[ -z "${ENV}" ]]; then
+  echo "Error: Missing Environment argument." >&2
+  echo "Usage: [BRANCH=<branch>] [REPO=<repo>] $0 <app> <env>" >&2
   exit 1
 fi
 
-if [[ -z "${APP}" ]];
-then
-  echo "Missing Application. Command: 'BRANCH=main ./bin/deploy <app> <env>'."
-  exit
+# 2. Validate VPN Connection
+# Check if the Nomad host is resolvable via DNS. Suppress command output.
+echo "Checking VPN connection to ${NOMAD_HOST}..."
+if ! nslookup "$NOMAD_HOST" >/dev/null 2>&1; then
+  echo "Error: Unable to resolve ${NOMAD_HOST}." >&2
+  echo "Please ensure you are connected to the VPN." >&2
+  exit 1
 fi
+echo "VPN check passed."
 
-if [[ -z "${APP}" ]];
-then
-  echo "Missing Environment. Command: 'BRANCH=main ./bin/deploy <app> <env>'."
-  exit
+# --- Get Git Commit SHA ---
+GITHUB_REPO_URL="https://github.com/${GITHUB_ORG}/${REPOSITORY}.git"
+echo "Fetching latest commit SHA for branch '${BRANCH_NAME}' from ${GITHUB_REPO_URL}..."
+
+# Use git ls-remote to get the ref, then awk to extract the full SHA.
+SHA=$(git ls-remote "${GITHUB_REPO_URL}" "${BRANCH_NAME}" | awk '{ print $1 }')
+
+# Validate that we got a SHA (should be 40 hex characters)
+if [[ -z "$SHA" || ${#SHA} -ne 40 ]]; then
+  echo "Error: Failed to retrieve a valid commit SHA for branch '${BRANCH_NAME}'." >&2
+  echo "Please check repository name ('${REPOSITORY}'), branch name ('${BRANCH_NAME}'), and network access." >&2
+  exit 1
 fi
+echo "Using commit SHA: ${SHA}"
+
+# --- Execute Remote Deployment ---
+echo "Deploying app '${APP}' env '${ENV}' using commit '${SHA}' via ${NOMAD_HOST}..."
 
-# Deploy using nomad-host-prod1, which has the nomad management key.
-SHA=$(git ls-remote https://github.com/pulibrary/${REPOSITORY}.git ${BRANCH_NAME} | awk '{ print substr($1,1,40) }')
-ssh deploy@nomad-host-prod1.lib.princeton.edu << EOF
-  curl -s "https://raw.githubusercontent.com/pulibrary/${REPOSITORY}/${SHA}/nomad/${APP}/deploy/${ENV}.hcl" | nomad job run -var "branch_or_sha=${SHA}" -
+# Construct the URL to the HCL file for the specific commit
+NOMAD_FILE_URL="https://raw.githubusercontent.com/${GITHUB_ORG}/${REPOSITORY}/${SHA}/nomad/${APP}/deploy/${ENV}.hcl"
+echo "Nomad job file URL: ${NOMAD_FILE_URL}"
+
+# Use SSH with a Here Document to execute the deployment command remotely.
+# curl fetches the HCL file, which is piped directly into 'nomad job run'.
+# The '-' argument tells nomad to read the job specification from standard input.
+# Variables ($NOMAD_FILE_URL, $SHA) are expanded *locally* before the command block is sent via SSH.
+if ! ssh "${SSH_USER}@${NOMAD_HOST}" <<EOF; then
```

Comment:
Something about the change here is making deploy not update the terminal numbers now, instead it's re-printing every time there's a change.